src/arch/src/aarch64/cache_info.rs

```diff
@@ -1,27 +1,43 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::fmt::{Display, Formatter};
-use std::path::{Path, PathBuf};
-use std::{fs, io, result};
+use std::{
+    any::Any,
+    fmt::{Display, Formatter},
+    fs, io,
+    path::{Path, PathBuf},
+    result,
+};
 
 use logger::warn;
 
 // Based on https://elixir.free-electrons.com/linux/v4.9.62/source/arch/arm64/kernel/cacheinfo.c#L29.
 const MAX_CACHE_LEVEL: u8 = 7;
 
 #[derive(Debug)]
-pub enum Error {
+pub(crate) enum Error {
     FailedToReadCacheInfo(io::Error),
     InvalidCacheAttr(String, String),
     MissingCacheConfig,
-    MissingOptionalAttr(String),
+    MissingOptionalAttr(String, CacheEntry),
 }
 
 type Result<T> = result::Result<T, Error>;
 
-#[derive(Clone)]
-pub(crate) struct CacheInfo {
+struct CacheEngine {
+    store: Box<dyn CacheStore>,
+}
+
+trait CacheStore: std::fmt::Debug {
+    fn get_by_key(&self, index: u8, file_name: &str) -> Result<String>;
+
+    fn as_any(&self) -> &dyn Any;
```

Comment:
are the `as_any` and `as_mut_any` methods used anywhere?
I commented them and the code still compiles

Reply:
It seems to be needed in tests to downcast the trait to an actual type. Anyway, we already have such a trait defined in [devices](https://github.com/firecracker-microvm/firecracker/blob/96d0ce9b97e7443125ccca9b89effe21720d7b15/src/devices/src/virtio/mod.rs#L72) . Can we move it to `utils` and reuse it here ?

Reply:
I rewrote the tests to get rid of the Any implementation.